CMakeLists.txt

```diff
@@ -11,23 +11,38 @@ if(NOT COMMAND add_llvm_library)
     endif()
     if (DEFINED ENV{LLVM_DIR})
         # We need to match the build environment for LLVM:
-        # In particular, we need C++11 and the -fno-rtti flag
+        # In particular, we need C++14 and the -fno-rtti flag
         set(CMAKE_CXX_STANDARD 14)
-	if(CMAKE_BUILD_TYPE MATCHES "Debug")
-		set(CMAKE_CXX_FLAGS "-fPIC -std=gnu++14 -g -O0 -fno-rtti -Wno-deprecated -DHAVE_IEEE_754 -DSIZEOF_VOID_P=8 -DSIZEOF_LONG=8")
-	else()
-		set(CMAKE_CXX_FLAGS "-fPIC -std=gnu++14 -O3 -fno-rtti -Wno-deprecated -DHAVE_IEEE_754 -DSIZEOF_VOID_P=8 -DSIZEOF_LONG=8")
-	endif()
-	set(CMAKE_C_FLAGS "-fPIC")
-
-	find_package(LLVM REQUIRED CONFIG)
+        set(CMAKE_CXX_STANDARD_REQUIRED ON)
+        # add -std=gnu++14
+        set(CMAKE_CXX_EXTENSIONS ON)
+
+        set(COMMON_FLAGS "-fPIC -Werror")
+        set(CMAKE_CXX_FLAGS  "${CMAKE_CXX_FLAGS} ${COMMON_FLAGS} -fno-rtti")
```

Comment:
Sorry for one more thing here. Lines 20-21 seems redundant if we have Lines 25-29. Could you merge them to make it compact?

Reply:
No. They are not redundant.
25-29 are only for setting up `-Ox`, while 21-22 are for setup options for C and CXX sources in common.